projects/packages/my-jetpack/_inc/utils/get-product-checkout-url.js

```diff
@@ -2,29 +2,20 @@
 /* global myJetpackInitialState */
 
 /**
- * External dependencies
- */
-import { getRedirectUrl } from '@automattic/jetpack-components';
-
-/**
- * Internal dependencies
- */
-import { MY_JETPACK_PRODUCT_CHECKOUT } from '../constants';
-
-/**
- * Return the product redirect URL, according to
- * the Jetpack redirect source, site, path, and redirect_to params.
+ * Return the checkout URL for the given product.
+ * It checkes whether the user is connected to Jetpack.
+ * It doesn't use Jetpack redirect because
+ * it's not possible to pass a valid redirect_to parameter.
+ * In short, and to make it work, it replaces the `=` character with `%3D`.
  *
- * @param {string} product - Checkout product name
+ * @param {string} product          - Checkout product name
+ * @param {boolean} isUserConnected - True when the user is connected Jetpack
  * @returns {string} the redirect URL
  */
-export default function getProductCheckoutUrl( product ) {
-	const { siteSuffix: site, redirectUrl } = window?.myJetpackInitialState || {};
-	const redirect_to = `${ redirectUrl }&product=${ product }`;
-
-	return getRedirectUrl( MY_JETPACK_PRODUCT_CHECKOUT, {
-		site,
-		path: 'jetpack_search',
-		query: `redirect_to=${ redirect_to }`,
-	} );
+export default function getProductCheckoutUrl( product, isUserConnected ) {
+	const { siteSuffix, redirectUrl } = window?.myJetpackInitialState || {};
+	const redirect_to = `${ redirectUrl.replace( /=/g, '%3d' ) }${
+		! isUserConnected ? '&unlinked=1' : ''
+	}&site=${ siteSuffix }`;
+	return `https://wordpress.com/checkout/${ siteSuffix }/${ product }?redirect_to=${ redirect_to }`;
```

Comment:
Did you try to use the `URL` utility from the browser to handle all of this?

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

It provides good handling of `searchParams`:

https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

Reply:
It doesn't seem to be completely supported by all browsers. Looking...

Reply:
> It provides good handling of searchParams:

> https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams

Is it to build or to stripe them? According to the doc:
> The searchParams readonly property of the [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) interface returns a [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) object allowing access to the [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) decoded query arguments contained in the URL.

Reply:
Just linking back to our discussion in slack 🧵  about Safari support and write access: p1644363459463389-slack-C02TQF5VAJD

Reply:
Addressed [here](https://github.com/Automattic/jetpack/pull/22740/commits/56265321e2d677934436c6d2ad19f8a7db6132fe).